Lib/unittest/async_case.py

```diff
@@ -89,8 +89,9 @@ def _callMaybeAsync(self, func, /, *args, **kwargs):
         else:
             return ret
 
-    async def _asyncioLoopRunner(self):
-        queue = self._asyncioCallsQueue
+    async def _asyncioLoopRunner(self, fut):
+        self._asyncioCallsQueue = queue = asyncio.Queue()
```

Comment:
Just curious why we moved it here. Is there a difference in constructing it in `_setupAsyncioLoop` itself without passing loop and having it here?

Reply:
Well, it is possible to just drop the `loop` parameter from `Queue` construction because we installed the default event loop a few lines above but I did two steps further.

Instantiation on asyncio loop coupled objects from an async function is more idiomatic, it always grabs a currently executed loop (the loop which is used to run a coroutine).
It prevents things like this:
```
import asyncio
queue = asyncio.Queue()

asyncio.run(...)  # a code that works with queue
```
In the example `queue` variable is instantiated with the default asyncio loop but `run()` creates a new loop version. So the queue doesn't work and hangs eventually because a future created by one loop cannot be awaited by another one.

In Python 3.9 I have a plan to add another deprecation warning if `not self._loop.is_running()` for queues and locks and eventually replace `get_event_loop()` with `get_running_loop()` as a final fix.